src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoader.cs

```diff
@@ -318,17 +340,28 @@ public sealed class Options
         /// </summary>
         internal readonly struct Segment
         {
+            public readonly string Name;
             public readonly int Min;
             public readonly int Lim;
             public readonly bool ForceVector;
 
             public Segment(int min, int lim, bool forceVector)
             {
                 Contracts.Assert(0 <= min & min < lim);
+                Name = null;
                 Min = min;
                 Lim = lim;
                 ForceVector = forceVector;
             }
+
+            public Segment(string name, bool forceVector)
+            {
+                Contracts.Assert(name != null);
+                Name = name;
+                Min = 0;
```

Comment:
Maybe `-1` would be better for these? Since that is an invalid value. Rather than using `0`, which could possibly work.